sdk/core/amqp-common/src/retry.ts

```diff
@@ -97,12 +97,22 @@ async function checkNetworkConnection(host: string): Promise<boolean> {
   if (isNode) {
     return new Promise((res) => {
       resolve(host, function(err: any): void {
-        // List of possible DNS error codes: https://nodejs.org/dist/latest-v12.x/docs/api/dns.html#dns_error_codes
-        if (err && (err.code === "ECONNREFUSED" || err.code === "ETIMEOUT")) {
-          res(false);
+        if (err) {
+          log.retry(
+            "Error thrown from dns.resolve in network connection check: '%s', %O",
+            err.code || err.name,
+            err
+          );
+
+          // List of possible DNS error codes: https://nodejs.org/dist/latest-v12.x/docs/api/dns.html#dns_error_codes
+          if (err.code === "ECONNREFUSED" || err.code === "ETIMEOUT") {
+            return res(false);
+          }
         } else {
-          res(true);
+          log.retry("Successfully resolved host via dns.resolve in network connection check");
         }
+
+        return res(true);
```

Comment:
It's not clear to me - why is it that any other error is considered 'ok'? It looks like we can fall through.

Reply:
I added a comment. In short, we only expect to see either of these two errors from dns.resolve if the network is actually down. Any other error could occur for other reasons.

It's not perfect but it's the closest we've gotten to determine whether we have a network connection or not...